### PR TITLE
fix: hide privacy policy toast during onboarding, and for onboarded users

### PR DESCRIPTION
**Pull request by jonybur**
## **Description**

Fixes two bugs:
- Privacy policy toast should not be seen during onboarding
- Users who onboarded after the privacy policy date should not see the new privacy policy toast.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25217?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

New user onbaording after the date:
1. Set the privacy policy date to a date in the future from today
2. Onboard (create a new user)
3. User should not see the privacy policy toast at any moment

New user onboarding before the date:
1. Set the privacy policy date to a date in the past from today.
2. Onboard (create a new user)
3. User should not see the privacy policy toast at any moment
4. Set the privacy policy date to a date in the future from today
5. User should not see the privacy policy toast on the home screen

Old user after the privacy policy date:
1. Have a user already onboarded
2. Make sure the `onboardingDate` is `null` in the redux state (this simulates a user who has registered before this PR has been in production)
3. Set the privacy policy date to a date in the future from today
4. User should not see the new privacy policy toast

Old user before the privacy policy date:
1. Have a user onboarded
2. Make sure the `onboardingDate` is `null` in the redux state (this simulates a user who has registered before this PR has been in production)
3. Set the privacy policy date to a date in the past before today
4. User should see the new privacy policy toast

TL;DR: If the privacy policy date is in the future, the user should never see it -BUT- if the privacy policy is in the past, the user should see it, as long as the user onboarded before the privacy policy date.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by salimtb** [comment]
is it posible to add an e2e test for this ?

File: ui/pages/onboarding-flow/onboarding-flow.js [hunk 1]
```diff
@@ -70,6 +71,10 @@ export default function OnboardingFlow() {
   const isFromReminder = new URLSearchParams(search).get('isFromReminder');
   const trackEvent = useContext(MetaMetricsContext);
 
+  useEffect(() => {
```
Comment (salimtb) [inline ui/pages/onboarding-flow/onboarding-flow.js:74]: can we add a comment on the top of this use effect to explain why we have it here ?

File: ui/store/actions.ts [hunk 2]
```diff
@@ -4250,6 +4250,12 @@ export function setNewPrivacyPolicyToastClickedOrClosed() {
   };
 }
 
+export function setOnboardingDate() {
```
Comment (salimtb) [inline ui/store/actions.ts:4253]: is it possible to add a unit test for this function on the `action.test.js` ? should be sample but better for the test coverage

**Review by bergeron — APPROVED** [review]

**Comment by bergeron** [comment]
I haven't seen a toast appear yet when following the steps in `New user onboarding before the date:`

**Review by bergeron — APPROVED** [review]
Ok with some minor code edits i think ive reproduced all the cases correctly

**Review by darkwing — APPROVED** [review]

**Review by salimtb — APPROVED** [review]
